pkg/certsigner/signer.go

```diff
@@ -125,29 +150,50 @@ func (s *CertServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	s.mux.ServeHTTP(w, r)
 }
 
-// NewSigner returns a CertSigner object after filling in its attibutes
-// from the `Config` provided.
-func NewSigner(c Config) (*CertSigner, error) {
-	ca, err := ioutil.ReadFile(c.CACertFile)
+// newSignerCA returns a SignerCA object of PEM encoded CA cert and keys based on the profile passed.
+func newSignerCA(sc *SignerCAFiles, profile string) (*SignerCA, error) {
+	if profile == "" {
+		return nil, ErrInvalidOrg
+	}
+	caCert := sc.CACert
```

Comment:
a switch might be more useful  in this change.

Reply:
Yep good point